File: codecheaf/CHFHEIST.py
for i in range(int(input())):
    D, d, p, q = map(int, input().split())
    vi = D // d
    if vi%2==0:
        s = d*vi//2*(2*p+(vi-1)*q)
        s+=(D%d)*(p+vi*q)
        print(s)
    else:
        s = d*(vi*(p+(vi-1)//2)*q)
        s+=(D%d)*(p+vi*q)
        print(s)    

File: codecheaf/CHFHEIST.cpp
#include <bits/stdc++.h>
#define ll long long int
#define INF 2e18
using namespace std;
int main()
{
#ifndef ONLINE_JUDGE
    freopen("input.txt", "r", stdin);
    freopen("output.txt", "w", stdout);
#endif

    ll tc;
    cin >> tc;
    while (tc--)
    {
        int D, d, p, q;
        cin >> D >> d >> p >> q;
        ll ans = 0;
        ll n = D / d;
        ans = (p * n + (q * (n - 1) * n) / 2) * d;
        if (D % d != 0)
        {
            ans += (D % d) * (p + n * q);
        }
        cout << ans << endl;
    }

    return 0;
}